In [1]:
import pandas as pd

In [2]:
df = pd.read_csv("../data/titanic_data.csv")

## Basic Information

In [3]:
df.head()

,PassengerId,Survived,Pclass,Name,Sex,Age,SibSp,Parch,Ticket,Fare,Cabin,Embarked
0,1,0,3,"Braund, Mr. Owen Harris",male,22.0,1,0,A/5 21171,7.2500,NaN,S
1,2,1,1,"Cumings, Mrs. John Bradley (Florence Briggs Th...",female,38.0,1,0,PC 17599,71.2833,C85,C
2,3,1,3,"Heikkinen, Miss. Laina",female,26.0,0,0,STON/O2. 3101282,7.9250,NaN,S
3,4,1,1,"Futrelle, Mrs. Jacques Heath (Lily May Peel)",female,35.0,1,0,113803,53.1000,C123,S
4,5,0,3,"Allen, Mr. William Henry",male,35.0,0,0,373450,8.0500,NaN,S


In [5]:
df.shape

(891, 12)

In [6]:
df.info()

<class 'pandas.DataFrame'>
RangeIndex: 891 entries, 0 to 890
Data columns (total 12 columns):
 #   Column       Non-Null Count  Dtype  
---  ------       --------------  -----  
 0   PassengerId  891 non-null    int64  
 1   Survived     891 non-null    int64  
 2   Pclass       891 non-null    int64  
 3   Name         891 non-null    str    
 4   Sex          891 non-null    str    
 5   Age          714 non-null    float64
 6   SibSp        891 non-null    int64  
 7   Parch        891 non-null    int64  
 8   Ticket       891 non-null    str    
 9   Fare         891 non-null    float64
 10  Cabin        204 non-null    str    
 11  Embarked     889 non-null    str    
dtypes: float64(2), int64(5), str(5)
memory usage: 83.7 KB


The dataset contains ``891 Titanic passenger records and 12 columns.`` it includes passenger details such as age, gender, claas, fare, family, size, cabin, and survival status. Some values are missing in `` Age, Cabin, and Embarked.`` The Survived column is the target variable

## Specific Information

In [7]:
# stat info: mean(Q1), median(Q2), Q3 mode std max corr
df[["Age", "Fare"]].mean()

Age     29.699118
Fare    32.204208
dtype: float64

In [8]:
df[["Age", "Fare"]].median()

Age     28.0000
Fare    14.4542
dtype: float64

In [9]:
df[["Age", "Fare"]].describe()

,Age,Fare
count,714.000000,891.000000
mean,29.699118,32.204208
std,14.526497,49.693429
min,0.420000,0.000000
25%,20.125000,7.910400
50%,28.000000,14.454200
75%,38.000000,31.000000
max,80.000000,512.329200


In [10]:
df.Embarked.mode()

0    S
Name: Embarked, dtype: str

**Description:**

- Average age of passengers is approximately 29.7 years, indicating that most passengers were relatively young.
- Age distribution is fairly symmetric, as the mean (29.7) and median (28.0) are close.
- The minimum age is 0.42 years (about 5 months), and the oldest passenger is 80 years old.
- There are missing values in the Age column (714 entries vs. 891 for Fare), suggesting some age data is not available.
- For Fare, the mean is 32.20 and the median is 14.45 — a large gap indicating a right-skewed (positively skewed) distribution.
- Fare values range from 0 (free tickets) to over 512, showing a wide spread and presence of outliers or high-paying passengers.
- The high standard deviation (49.69) also supports the presence of large variability in fare prices.

In [11]:
# Counts

df["Embarked"].value_counts(normalize= False, dropna= True)

Embarked
S    644
C    168
Q     77
Name: count, dtype: int64

In [12]:
df.value_counts("Embarked", normalize= False, dropna= True)

Embarked
S    644
C    168
Q     77
Name: count, dtype: int64

In [13]:
df.Embarked.value_counts(normalize= False, dropna= True)

Embarked
S    644
C    168
Q     77
Name: count, dtype: int64

Interestingly, large group of passangers embarked from southampton

In [15]:
df.value_counts(["Sex", "Survived"])

Sex     Survived
male    0           468
female  1           233
male    1           109
female  0            81
Name: count, dtype: int64

In [19]:
df[df.Sex == "female"].value_counts("Survived")

Survived
1    233
0     81
Name: count, dtype: int64

In [20]:
# Embarked from S, Fare Distribution ?
df[df.Embarked == "S"]["Fare"].describe()

count    644.000000
mean      27.079812
std       35.887993
min        0.000000
25%        8.050000
50%       13.000000
75%       27.900000
max      263.000000
Name: Fare, dtype: float64

In [22]:
# Embarked from S and gender Male , Survival rate ?

df[
    (df.Embarked == "S") & (df["Sex"] == "male")
].value_counts("Survived")

Survived
0    364
1     77
Name: count, dtype: int64

In [25]:
# Embarked from S and gender female , Survival rate ?
df[
    (df.Embarked == "S") & (df["Sex"] == "female")
].value_counts("Survived")

Survived
1    140
0     63
Name: count, dtype: int64